azure-kusto-ingest/azure/kusto/ingest/base_ingest_client.py

```diff
@@ -78,6 +78,13 @@ def ingest_from_stream(self, stream_descriptor: Union[StreamDescriptor, IO[AnySt
         """
         pass
 
+    @abstractmethod
+    def set_proxy(self, proxy_url: str):
```

Comment:
I think it's better to set it through optional ctor parameter

Reply:
In the PR for the client you convinced me to not make it a ctor parameter - https://github.com/Azure/azure-kusto-python/pull/360/files/efde7dd3245bfec121eaf4b5958bbbdac3333557#diff-ed58329617a49a8000e9b4c91a78ad8be106646e8f41e37403d57e2dc48f09c5

What's different here?

Reply:
The difference is the proxy is a setting of the KustoClient, and here you set a property of a member - if we didn't need to use the proxy for storage as well, I'd say we should add a way to create a client with KustoClient in addition to creation with a uri or kcsb